src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_DesignerAttributes.cs

```diff
@@ -17,16 +19,22 @@ internal partial class CodeAnalysisService : IRemoteDesignerAttributeService
         /// 
         /// This will be called by ServiceHub/JsonRpc framework
         /// </summary>
-        public async Task<ImmutableArray<DesignerAttributeDocumentData>> ScanDesignerAttributesAsync(ProjectId projectId, CancellationToken cancellationToken)
+        public async Task<IReadOnlyList<DesignerAttributeDocumentData>> ScanDesignerAttributesAsync(ProjectId projectId, CancellationToken cancellationToken)
         {
             using (RoslynLogger.LogBlock(FunctionId.CodeAnalysisService_GetDesignerAttributesAsync, projectId.DebugName, cancellationToken))
             {
                 var solution = await GetSolutionAsync(cancellationToken).ConfigureAwait(false);
                 var project = solution.GetProject(projectId);
+
                 var data = await AbstractDesignerAttributeService.TryAnalyzeProjectInCurrentProcessAsync(
                     project, cancellationToken).ConfigureAwait(false);
 
-                return data.Values.ToImmutableArray();
+                if (data.Count == 0)
+                {
+                    return SpecializedCollections.EmptyReadOnlyList<DesignerAttributeDocumentData>();
+                }
+
+                return data.Values.ToList();
```

Comment:
I feel like i should be seeing more of this in the code.  Basically, all the places where we return an immutable array (even casted to an IReadOnlyList) would need to actually be converted to a List<T>.  THat's because json.net isn't going to see or care that this is IReadOnlyList.  It's going to look at the real underlying type and will decide what to do with it.  If that underlying type is stilL IMmutableArray, things will still not work.